formula/src/main/java/com/instacart/formula/Action.kt

```diff
@@ -0,0 +1,121 @@
+package com.instacart.formula
+
+/**
+ * A deferred action returned by [evaluation][Formula.evaluate] that will run for any new unique
+ * value of [key] and will be be cleaned up if [key] changes or if [Action] is
+ * left out of evaluation.
+ *
+ * Action can produce and send [events][Event] back to the [Formula] instance. It
+ * can be used to subscribe to RxJava observables, Kotlin Flows, event bus, or any other event
+ * mechanism.
+ *
+ * An [Action] key is a value that defines the identity of the [Action]. If
+ * a key changes, the [Action] will dispose old instance and run again from a fresh state.
+ * We also include the code call-site as a key parameter (concept called positional memoization).
+ *
+ * To construct a RxJava based action (using formula-rxjava3 library), you can do the following:
+ *
+ * ```kotlin
+ * val action = RxAction.fromObservable { Observable.just(1, 2, 3) }
+ * ```
+ *
+ * To use it within a [Formula]:
+ * ```
+ * Evaluation(
+ *   actions = context.actions {
+ *     action.onEvent { event ->
+ *       transition()
+ *     }
+ *   }
+ * )
+ * ``
+ *
+ * @param Event A type of event used to notify [Formula].
+ */
+interface Action<Event> {
```

Comment:
Copy pasted from `Stream` with big documentation changes

Reply:
Still see `Stream` in the docs below.  Intentional?

Reply:
Good catch, updated